tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -887,6 +885,58 @@ private async Task UploadTestRunBlobAsync(string account, Build build, TestRun t
             }
         }
 
+        private async Task UploadTestRunResultBlobAsync(string account, Build build, TestRun testRun)
+        {
+            try
+            {
+                var continuationToken = string.Empty;
+                var buildIds = new[] { build.Id };
+                var discoveredTestResults = new List<TestCaseResult>();
+                var blobPath = $"{build.Project.Name}/{testRun.CompletedDate:yyyy/MM/dd}/{testRun.Id}.jsonl";
+                var blobClient = this.testResultContainerClient.GetBlobClient(blobPath);
+
+                if (await blobClient.ExistsAsync())
+                {
+                    this.logger.LogInformation("Skipping existing test results blob for build {BuildId}, test run {RunId}", build.Id, testRun.Id);
+                    return;
+                }
+
+                var data = await testResultsClient.GetTestResultsAsync(build.Project.Id, testRun.Id);
+                
+                var content = JsonConvert.SerializeObject(data.Select(x => new
```

Comment:
For json blobs with multiple entites, rather than writing a json array to the blob, we write whole objects as individual lines to the blob i.e. `jsonl` vs `json`

json:
```json
[
  {},
  {}
]
```

jsonl:
```json
{}
{}
```


Kusto parses each line individually, so no bounding square brackets or trailing commas are needed

```cs
var content = new StringBuilder();
foreach (var record in results)
{
    builder.AppendLine(JsonConvert.SerializeObject(new {
       ...
```

Reply:
For the timeline records, it's a small set that we already have in memory, so we just serialize each entry onto a stringbuilder and write the stringbuilder's contents to the blob.  I think we should do the same here.

https://github.com/Azure/azure-sdk-tools/blob/main/tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs#L626-L672